projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -0,0 +1,252 @@
+    private readonly inputPhone?: TuiPrimitiveTextfieldComponent;
+
+    @ViewChild(TuiTextfieldComponent)
+    private readonly countrySelect?: TuiTextfieldComponent<unknown>;
+
+    @Input()
+    public countries = this.options.countries;
+
+    @Output()
+    public readonly countryIsoCodeChange = this.countryIsoCode$.pipe(
+        distinctUntilChanged(),
+        skip(1),
+    );
+
+    @Input('countryIsoCode')
+    public set isoCode(code: TuiCountryIsoCode) {
+        this.countryIsoCode$.next(code);
+    }
+
+    public get nativeFocusableElement(): HTMLElement | null {
+        return this.inputPhone && !this.computedDisabled
+            ? this.inputPhone.nativeFocusableElement
+            : null;
+    }
+
+    public get focused(): boolean {
+        return Boolean(this.countrySelect?.focused || this.inputPhone?.focused);
+    }
+
+    public onPaste(event: Event, phonesMetadata: MetadataJson | null): void {
+        if (
+            !tuiIsInputEvent(event) ||
+            !phonesMetadata ||
+            (!event.inputType.includes('Drop') && !event.inputType.includes('Paste'))
+        ) {
+            return;
+        }
+
+        const newValue = event.data || '';
+        const prefixedValue = newValue.startsWith(CHAR_PLUS)
+            ? newValue
+            : CHAR_PLUS + newValue;
+
+        if (validatePhoneNumberLength(prefixedValue) === 'TOO_SHORT') {
+            return;
+        }
+
+        const countryIsoCode = maskitoGetCountryFromNumber(prefixedValue, phonesMetadata);
+
+        if (countryIsoCode) {
+            this.countryIsoCode$.next(countryIsoCode);
+        }
+    }
+
+    public onItemClick(isoCode: TuiCountryIsoCode): void {
+        this.open = false;
+        this.countryIsoCode$.next(isoCode);
+        this.nativeFocusableElement?.focus();
+    }
+
+    constructor() {
+        super();
+        combineLatest([this.countryIsoCode$, this.options.metadata])
+            .pipe(
+                map(args => this.computeMask(...args)),
+                takeUntilDestroyed(),
+            )
+            .subscribe(this.mask$);
+    }
+
+    public override setDisabledState(): void {
+        super.setDisabledState();
+        this.close();
+    }
+
+    public override writeValue(unmaskedValue: string): void {
+        super.writeValue(unmaskedValue);
+
+        const maskOptions = this.mask$.value;
+
+        this.textfieldValue = maskOptions
+            ? maskitoTransform(unmaskedValue, maskOptions)
+            : unmaskedValue; // it will be calibrated later when mask is ready (by maskitoInitialCalibrationPlugin)
+        this.cdr.detectChanges();
+    }
+
+    protected onActiveZone(active: boolean): void {
+        this.updateFocused(active);
+    }
+
+    protected onFocus(phonesMetadata: MetadataJson): void {
+        const countryCallingCode = `${CHAR_PLUS + getCountryCallingCode(this.countryIsoCode$.value, phonesMetadata)} `;
+
+        if (!this.textfieldValue) {
```

Comment:
Why only when its empty?

Reply:
In all other cases country calling code already exists (no need to add it).